flake.nix

```diff
@@ -64,6 +66,7 @@
             ];
           ZSTD_SYS_USE_PKG_CONFIG = "1";
           LIBSSH2_SYS_USE_PKG_CONFIG = "1";
+          NIX_JJ_GIT_HASH = self.rev or "";
```

Comment:
Maybe `... or "<UNKNOWN>"`, or some other sentinel value that can be looked up in the code would be useful, here. Maybe it could be:

- `"<UNKNOWN>"`
- `"<UNKNOWN GIT HASH>"`

Or something along those lines. For reference, you *can* do stuff like use `tarball+https://` URIs to reference a flake ([reference](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake.html)), so that's a way to install `jj` which I don't think this case would handle. I don't think most people would do this, but some do.

Reply:
If you have a dirty git state (so you check out and then edit some stuff), `self.rev` would not be set if you run `nix build`.

But if you run `cargo build` the git head would be there anyway, which would not be correct technically - if you have a git repo anyway, it first looks up the working copy commit of a jj repo which avoids the issue by not having dirty state at all :)

For now I'll just merge this for the initial reason of `nix profile install github:martinvonz/jj` not having the revision in `jj -V` right now, but perhaps this might need to be revisited again